packages/core-plugin-api/src/icons/types.ts

```diff
@@ -36,10 +36,10 @@ import { ComponentType } from 'react';
 export type IconComponent = ComponentType<
   /* Material UI v4 */
   | {
-      fontSize?: 'large' | 'small' | 'default';
+      fontSize?: 'large' | 'small' | 'default' | 'inherit';
     }
   /* Material UI v5: https://mui.com/material-ui/migration/v5-component-changes/#icon */
   | {
-      fontSize?: 'medium' | 'large' | 'small';
+      fontSize?: 'medium' | 'large' | 'small' | 'inherit';
```

Comment:
Noting that strictly speaking this is a breaking change, but given the assumption on this being backed by a MUI icon component in practice I think it's alright to ship